subject_level/fmri_ants_bids.py

```diff
@@ -291,6 +291,150 @@ def create_reg_workflow(name='registration'):
+                                                          'topup_fieldcoef',
+                                                          'topup_movpar',
+                                                          'topup_corrected',
+                                                          'applytopup_corrected'
+                                                          ]),
+                      name='outputspec')
+
+    pe_dirs = {'j-':-1, 'j':1} #AP:j-, PA:j
+    
+    opp_pe_dir = [pe_dir for pe_dir in pe_dirs.keys() if pe_dir != orig_pe_dir][0]
+
+    topup2median = Node(fsl.FLIRT(out_file='orig2median.nii.gz', 
+                                  output_type='NIFTI_GZ', interp='spline'), 
+                        name='orig2median')
+    topup2median.inputs.dof = 6
+    topup2median.inputs.out_matrix_file = 'orig2median'
+
+    applyxfm = Node(fsl.ApplyXfm(out_file='opp2median.nii.gz' 
+                                 apply_xfm=True, interp='spline', output_type='NIFTI_GZ'),
+                    name='applyxfm')        
+    topup.connect(topup2median, 'out_matrix_file', applyxfm, 'in_matrix_file')
+
+    make_topup_list = Node(Merge(2), name='make_topup_list')
+    topup.connect(topup2median, 'out_file', make_topup_list, 'in1')
+    topup.connect(applyxfm, 'out_file', make_topup_list, 'in2')
+
+    merge_topup = Node(fsl.Merge(dimension='t', output_type='NIFTI_GZ'), 
+                        name='merge_topup')
+    topup.connect(make_topup_list, 'out', merge_topup, 'in_files')
+
+    file_writer_topup = Node(Function(input_names=['readout', 'fname',
+                                                   'direction'],
+                                output_names=['encoding_file'],
+                                function=write_encoding_file),
+                       name='file_writer_topup')
+    file_writer_topup.inputs.readout = readout_topup
+    file_writer_topup.inputs.fname = 'topup'
+    file_writer_topup.inputs.direction = pe_dirs[orig_pe_dir]
+
+    run_topup = Node(fsl.TOPUP(out_corrected='b0correct.nii.gz', numprec='float', 
+                        config='b02b0.cnf', output_type='NIFTI_GZ'), 
+                    name='run_topup')
+    topup.connect(file_writer_topup, 'encoding_file', run_topup, 'encoding_file')
+
+    applytopup = Node(fsl.ApplyTOPUP(output_type='NIFTI_GZ'), name='applytopup')
+    applytopup.inputs.in_index = [1]
+    applytopup.inputs.method = 'jac'   
+
+    file_writer_ts = file_writer_topup.clone(name='file_writer_ts')
+    file_writer_ts.inputs.readout = readout
+    file_writer_ts.inputs.fname = 'bold_ts'
+
+    topup.connect(merge_topup, 'merged_file', run_topup, 'in_file')
+    topup.connect(file_writer_ts, 'encoding_file', applytopup, 'encoding_file')
+    topup.connect(run_topup, 'out_fieldcoef', applytopup, 'in_topup_fieldcoef')
+    topup.connect(run_topup, 'out_movpar', applytopup, 'in_topup_movpar')
+
+    if num_slices % 2 != 0:
```

Comment:
since i'm calculating this in a nipype node, i'm unsure how to feed it in here - other thought are making a mini-workflow just for odd # of slices